index.js

```diff
@@ -573,8 +574,23 @@ class Raspi extends EventEmitter {
     throw new Error('reportDigitalPin is not yet implemented');
   }
 
-  pingRead() {
-    throw new Error('pingRead is not yet implemented');
+  pingRead(config, handler) {
+    const pulseOut = config.pulseOut;
+    const pin = config.pin;
+    const sonar = new Sonar(config.pin);
+    
+    const pinInstance = this[getPinInstance](this.normalize(pin));
+    if (pinInstance.mode != INPUT_MODE) {
+      this.pinMode(pin, INPUT_MODE);
```

Comment:
Just to confirm my understanding of all the corrections:

**New pin mode**
```javascript
const PING_READ_MODE = 98;
```

**Update pingRead to make use of the new mode***
```javascript
const pinInstance = this[getPinInstance](this.normalize(pin));
if (pinInstance.mode != PING_READ_MODE) {
  this.pinMode(pin, PING_READ_MODE);
}
```
I'm not sure about the below line though. 

>There's one catch though, we'll have to set the mode on the pinInstance to UNKNOWN_MODE though inside of the pinMode method, since Johnny-Five won't recognize PING_READ_MODE.

Based on @rwaldron 's update to [pingRead settings handler](https://github.com/rwaldron/io-plugins#pingreadsettings-handler), it read "No pin mode specified". So my question is, do we need to still worry about the catch? And if so, do you mean something like the below? (sorry if i'm completely off track here).

```javascript
pinMode(pin, mode) {
if (mode === PING_READ_MODE) //as johnny-five can't recognize PING_READ_MODE
  mode = UNKNOWN_MODE;
      
  this[pinMode]({ pin, mode });
}
```

**Use `pinInstance.peripheral` to hold the reference to an instance of Sonar**
```javascript
if (pin == LED_PIN && !(pinInstance.peripheral instanceof LED)) {
  pinInstance.peripheral = new LED();
} else {
  switch (mode) {
	case INPUT_MODE:
	  pinInstance.peripheral = new DigitalInput(config);
	  break;
	case OUTPUT_MODE:
	  pinInstance.peripheral = new DigitalOutput(config);
	  break;
	case PWM_MODE:
	case SERVO_MODE:
	  pinInstance.peripheral = new PWM(normalizedPin);
	  break;
	case PING_READ_MODE:  //<-- new code here
	  pinInstance.peripheral = new Sonar(normalizedPin);
	  break;
	default:
	  console.warn('Unknown pin mode: ' + mode);
	  break;
  }
}
```